pkg/resources/kafka/pod.go

```diff
@@ -342,22 +342,53 @@ func generatePodAntiAffinity(clusterName string, hardRuleEnabled bool) *corev1.P
 	return &podAntiAffinity
 }
 
-func generateDataVolumeAndVolumeMount(pvcs []corev1.PersistentVolumeClaim) (volume []corev1.Volume, volumeMount []corev1.VolumeMount) {
+func generateDataVolumeAndVolumeMount(pvcs []corev1.PersistentVolumeClaim, storageConfigs []v1beta1.StorageConfig) ([]corev1.Volume, []corev1.VolumeMount) {
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+
+	// PVC volume mount (pvcs are already sorted by pvc name in 'getCreatedPvcForBroker' function
 	for i, pvc := range pvcs {
-		volume = append(volume, corev1.Volume{
+		volumes = append(volumes, corev1.Volume{
 			Name: fmt.Sprintf(kafkaDataVolumeMount+"-%d", i),
 			VolumeSource: corev1.VolumeSource{
 				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
 					ClaimName: pvc.Name,
 				},
 			},
 		})
-		volumeMount = append(volumeMount, corev1.VolumeMount{
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
 			Name:      fmt.Sprintf(kafkaDataVolumeMount+"-%d", i),
 			MountPath: pvc.Annotations["mountPath"],
 		})
 	}
-	return
+
+	// emptyDir volume mounts
+	var emptyDirs []v1beta1.StorageConfig
+	for i := range storageConfigs {
+		if storageConfigs[i].EmptyDir != nil {
```

Comment:
When we have a storageConfig with a PvcSpec!=nil and EmptyDir!=nil then I think there will be volume duplications. PvcSpec has higher priority.